libraries/griptape_nodes_library/griptape_nodes_library/execution/for_each_start.py

```diff
@@ -0,0 +1,136 @@
+from typing import Any
+
+from griptape_nodes.exe_types.core_types import (
+    ControlParameterInput,
+    ControlParameterOutput,
+    Parameter,
+    ParameterList,
+    ParameterMode,
+    ParameterTypeBuiltin,
+)
+from griptape_nodes.exe_types.flow import ControlFlow
+from griptape_nodes.exe_types.node_types import BaseNode, EndLoopNode, StartLoopNode
+
+
+class ForEachStartNode(StartLoopNode):
+    """For Each Start Node that runs a connected flow for each item in a parameter list.
+
+    This node iterates through each item in the input list and runs the connected flow for each item.
+    It provides the current item to the next node in the flow and keeps track of the iteration state.
+    """
+
+    _items: list[Any]
+    _flow: ControlFlow | None = None
+
+    def __init__(self, name: str, metadata: dict[Any, Any] | None = None) -> None:
+        super().__init__(name, metadata)
+        self.finished = False
+        self.current_index = 0
+        self._items = []
+        self.exec_out = ControlParameterOutput(tooltip="Continue the flow", name="exec_out")
```

Comment:
Can we rename the "Flow Out" label to "For Each" to clarify for users what this is for?